ui/src/app/edge/history/consumption/widget.component.ts

```diff
@@ -28,8 +30,8 @@ export class ConsumptionComponent extends AbstractHistoryWidget implements OnIni
         super(service);
     }
 
-    ngOnInit() {
-        this.service.setCurrentComponent('', this.route).then(response => {
+    async ngOnInit() {
```

Comment:
a function call that you are not controlling shouldnt be asynchronous, did you find this somewhere or did you come up with that?

Reply:
is it guaranteed that the lifecycles will be called like expected, when they are async